src/include/homestore/btree/detail/btree_remove_impl.ipp

```diff
@@ -76,15 +82,20 @@ retry:
         ASSERT_IS_VALID_INTERIOR_CHILD_INDX(found, idx, my_node);
         end_idx = start_idx = idx;
     } else if constexpr (std::is_same_v< ReqT, BtreeRangeRemoveRequest< K > >) {
-        const auto count = my_node->template get_all< V >(req.next_range(), UINT32_MAX, start_idx, end_idx);
-        BT_NODE_REL_ASSERT_NE(count, 0, my_node, "get_all returns 0 entries for interior node is not valid pattern");
+        const auto count = my_node->template get_all< K, V >(req.next_range(), UINT32_MAX, start_idx, end_idx);
+        if (!count) {
+            ret = btree_status_t::not_found;
+            go_to_out = true;
+        }
+        //        BT_NODE_REL_ASSERT_NE(count, 0, my_node, "get_all returns 0 entries for interior node is not valid
+        //        pattern");
 
     } else if constexpr (std::is_same_v< ReqT, BtreeRemoveAnyRequest< K > >) {
         const auto count = my_node->template get_all< V >(req.m_range, UINT32_MAX, start_idx, end_idx);
         BT_NODE_REL_ASSERT_NE(count, 0, my_node, "get_all returns 0 entries for interior node is not valid pattern");
         end_idx = start_idx = (end_idx - start_idx) / 2; // Pick the middle, TODO: Ideally we need to pick random
     }
-
+    if (go_to_out) { goto out_return; }
```

Comment:
why do we need the go_to_out variable, why can't line 88, be just `goto out_return` ?

Reply:
Technically your suggestion is legit. The way that is implemented is due to compilation warning. If I implemented as you suggest, a complation warning would be thrown out that not the label out_retrun has not reached in some code path. 

error: label ‘out_return’ defined but not used [-Werror=unused-label
cc1plus: all warnings being treated as errors